xmlparser/src/main/scala/com/drewhk/stream/xml/Xml.scala

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.drewhk.stream.xml
+
+import akka.NotUsed
+import akka.stream.scaladsl.Flow
+import akka.stream.{ Attributes, FlowShape, Inlet, Outlet }
+import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
+import akka.util.ByteString
+import com.fasterxml.aalto.{ AsyncByteArrayFeeder, AsyncXMLInputFactory, AsyncXMLStreamReader }
+import com.fasterxml.aalto.stax.InputFactoryImpl
+
+import scala.annotation.tailrec
+import scala.collection.immutable
+
+object Xml {
+
+  /**
+   * XML parsing events emitted by the parser flow. These roughly correspond to Java XMLEvent types.
+   */
+  sealed trait ParseEvent
+  sealed trait TextEvent extends ParseEvent {
+    def text: String
+  }
+
+  case object StartDocument extends ParseEvent
+  case object EndDocument extends ParseEvent
+  final case class StartElement(localName: String, attributes: Map[String, String]) extends ParseEvent
+  final case class EndElement(localName: String) extends ParseEvent
+  final case class Characters(text: String) extends TextEvent
+  final case class ProcessingInstruction(target: Option[String], data: Option[String]) extends ParseEvent
+  final case class Comment(text: String) extends ParseEvent
+  final case class CData(text: String) extends TextEvent
+
+  /**
+   * Parser Flow that takes a stream of ByteStrings and parses them to XML events similar to SAX.
+   */
+  val parser: Flow[ByteString, ParseEvent, NotUsed] =
+    Flow.fromGraph(new StreamingXmlParser)
+
+  /**
+   * A Flow that transforms a stream of XML ParseEvents. This stage coalesces consequitive CData and Characters
+   * events into a single Characters event or fails if the buffered string is larger than the maximum defined.
+   */
+  def coalesce(maximumTextLength: Int): Flow[ParseEvent, ParseEvent, NotUsed] =
+    Flow.fromGraph(new Coalesce(maximumTextLength))
+
+  /**
+   * A Flow that transforms a stream of XML ParseEvents. This stage filters out any event not corresponding to
+   * a certain path in the XML document. Any event that is under the specified path (including subpaths) is passed
+   * through.
+   */
+  def subslice(path: immutable.Seq[String]): Flow[ParseEvent, ParseEvent, NotUsed] =
```

Comment:
it seems as if adding JavaDSL would actually be very simple?
just `getSubslice(List)`, `getCoalesce()`, `getParser` - done?